### PR TITLE
Allow to specify symbol extensions which are used by stacktrace resolving

### PR DESCRIPTION
**Pull request by todi1856**
…nissues, add UI to specify symbol extension## **Please read and consider what information you want to pass on to people reviewing this PR**

### Purpose of PR
**Type of change:**
- [X] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe) 

* **Description of the feature**
Stacktrace Utility was failing if symbol file had an extension .so.sym, .so.dbg. T
his PR allows you to customize symbol extensions, thus user can choose the extension which to use for symbol resolving
* **Links to screenshots, design docs, user docs, etc.**
* logcat-settings-symbol-extensions
* https://github.com/Unity-Technologies/com.unity.mobile.logcat/blob/custom-symbol-extensions/com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md
---
### Testing status

Testing scenario:
* Build & Run ForceCrash project with public symbols
* Extract symbols.zip and rename symbols file to extension .so.sym
* Crash the app
* Try to resolve stackrace
* It will succeed now
* Go to Logcat preferences, remove .so.sym from symbol extensions
* Resolve the stacktrace again, resolving will fail, since we're not looking for symbols with extension .so.sym

### CODE REVIEW

**Review by aurimasc — APPROVED** [review]

File: com.unity.mobile.android-logcat/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
  - Added inputs window - Tools->Window->Inputs, where you can easily inject keyboard/text input on to android device.
  - Add Scroll option for messages which allows you to explicitly to always scroll to end without automatic logic.
  - Device selection dropdown will contain device name + id, if device name is not available, only id will be displayed.
+ - Add UI for specifying symbol file extensions which are used during stacktrace resolving while looking for symbols
```
Comment (anjalin14) [inline com.unity.mobile.android-logcat/CHANGELOG.md:13]: You can now specify symbol file extensions in the new settings, Preferences->Android Logcat Settings> **Symbol Extensions**. Stacktrace Utility tool uses the symbol file extensions while looking for symbols to resolve stacktraces.

File: com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md [hunk 2]
```diff
@@ -61,5 +61,9 @@ From the regular expressions list, the one at the top is used for resolving the
 For more information, refer to [configure stacktrace regular expressions](stacktrace-utility-set-up.md#configure-stacktrace-regular-expressions).
 
 
+## Symbol Extensions
 
+![](images/logcat-settings-symbol-extensions.png)
 
+Use the **Symbol Extensions** setting to configure file extensions which are appended to symbol file names during stacktrace resolving.
+For example, if symbol name is **libunity** and the specified symbol extensions are **.so.sym**, **.so.dbg**, stacktrace utility will look for **libunity.so.sym** and **libunity.so.dbg** symbols.
```
Comment (anjalin14) [inline com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md:69]:
```suggestion
For example, if the symbol file name is `libunity` and the specified symbol extensions are `.so.sym`, `.so.dbg`, the Stacktrace Utility tool looks for `libunity.so.sym` and `libunity.so.dbg` symbols..
```

File: com.unity.mobile.android-logcat/Editor/AndroidLogcatSettingsProvider.cs [hunk 3]
```diff
@@ -15,13 +15,16 @@ class Styles
             public static GUIContent font = new GUIContent("Font", "Font used for displaying messages");
             public static GUIContent fontSize = new GUIContent("Font Size");
             public static GUIContent stactraceRegex = new GUIContent("Stacktrace Regex", "Configure regex used for resolving function address and library name");
+            public static GUIContent symbolExtensions = new GUIContent("Symbol Extensions", "Configure which file extension to append while looking for symbol files.");
```
Comment (anjalin14) [inline com.unity.mobile.android-logcat/Editor/AndroidLogcatSettingsProvider.cs:18]:
```suggestion
            public static GUIContent symbolExtensions = new GUIContent("Symbol Extensions", "Specify which file extension to append while looking for symbol files.");
```

File: com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md [hunk 4]
```diff
@@ -61,5 +61,9 @@ From the regular expressions list, the one at the top is used for resolving the
 For more information, refer to [configure stacktrace regular expressions](stacktrace-utility-set-up.md#configure-stacktrace-regular-expressions).
 
 
+## Symbol Extensions
 
+![](images/logcat-settings-symbol-extensions.png)
 
+Use the **Symbol Extensions** setting to configure file extensions which are appended to symbol file names during stacktrace resolving.
```
Comment (anjalin14) [inline com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md:68]:
```suggestion
Use the **Symbol Extensions** setting to specify file extensions which are appended to symbol file names during stacktrace resolving.

```

**Review by anjalin14 — APPROVED** [review]

File: com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md (hunk 2 above)
Comment (anjalin14) [reply com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md:69]: I think I've added an extra full-stop in the end. Please can you remove it. Thanks

File: com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md (hunk 4 above)
Comment (anjalin14) [reply com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md:68]: Could you please also mention Symbol Extensions in the intro paragraph where we say, "The settings are grouped under the following sections:"
- Symbol Extensions(#symbol-exensions)

Comment (todi1856) [reply com.unity.mobile.android-logcat/Documentation~/android-logcat-settings.md]: [`05aa0b8` (#177)](https://github.com/Unity-Technologies/com.unity.mobile.logcat/pull/177/commits/05aa0b82f3493fcbda7fc5a80fc1e667035faa62)

File: com.unity.mobile.android-logcat/CHANGELOG.md (hunk 1 above)
Comment (todi1856) [reply com.unity.mobile.android-logcat/CHANGELOG.md]: https://github.com/Unity-Technologies/com.unity.mobile.logcat/pull/177/commits/05aa0b82f3493fcbda7fc5a80fc1e667035faa62

**Review by IGuscin — APPROVED** [review]
Looks good, didn't encounter any issues while testing.

Tested with public/debug symbols, covered couple different extensions (.so/so.sym/so.dbg/sym.so/dbg.so/soso/file without extension), covered adding removing part/all extensions from the list.
Thanks for adding most common extensions to a predefined list.

**Pull request merged**